platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -158,6 +161,66 @@ void spu_regions_flash_config_non_secure_callable(uint32_t start_addr,
 		1 /* Lock */);
 }
 
+uint32_t spu_regions_flash_get_base_address_in_region(uint32_t region_id)
+{
+	return FLASH_BASE_ADDRESS +
+		((region_id - FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID) *
+			FLASH_SECURE_ATTRIBUTION_REGION_SIZE);
+}
+
+uint32_t spu_regions_flash_get_last_address_in_region(uint32_t region_id)
+{
+	return FLASH_BASE_ADDRESS +
+		((region_id - FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID + 1) *
+			FLASH_SECURE_ATTRIBUTION_REGION_SIZE) - 1;
+}
+
+uint32_t spu_regions_flash_get_start_id(void) {
+
+	return FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID;
+}
+
+uint32_t spu_regions_flash_get_last_id(void) {
+
+	return FLASH_SECURE_ATTRIBUTION_REGIONS_START_ID +
+		NUM_FLASH_SECURE_ATTRIBUTION_REGIONS - 1;
+}
+
+uint32_t spu_regions_flash_get_region_size(void) {
+
+	return FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+}
+
+uint32_t spu_regions_sram_get_base_address_in_region(uint32_t region_id)
+{
+	return SRAM_BASE_ADDRESS +
+		((region_id - SRAM_SECURE_ATTRIBUTION_REGIONS_START_ID) *
+			SRAM_SECURE_ATTRIBUTION_REGION_SIZE);
+}
+
+uint32_t spu_regions_sram_get_last_address_in_region(uint32_t region_id)
+{
+	return SRAM_BASE_ADDRESS +
+		((region_id - SRAM_SECURE_ATTRIBUTION_REGIONS_START_ID + 1) *
+			SRAM_SECURE_ATTRIBUTION_REGION_SIZE) - 1;
+}
+
+uint32_t spu_regions_sram_get_start_id(void) {
+
+	return SRAM_SECURE_ATTRIBUTION_REGIONS_START_ID;
+}
+
+uint32_t spu_regions_sram_get_last_id(void) {
```

Comment:
Consider adding a `spu_region_id_is_valid()` which checks that the id is within the FLASH or SRAM region.